### ISSUE TITLE
Can't rewrite last character

### ISSUE THREAD

**Issue by Kolos65** [hrutvikpatel/react-native-character-input#10]
If you fill the example inputs with random characters and then try to rewrite it with something else, the last character input is treated differently and you are unable to change the previously given value. Something might be wrong in the traverseInputs function.

**Comment by Kolos65** [comment]
This was the problem:
```
const clearInputOnFocus = (inputPos: number) => {
    if (inputPos !== inputLength - 1) {
      singleInputRef[inputPos].clear();
      updateChangedChar(inputPos, '');
    }
  }
```
Is there a reason why the last character was not cleared on focus?

**Comment by hrutvikpatel** [comment]
Hey, thank you for finding this bug. Would you be able to send me a snippet of the code you used with this component, such that I can debug it and fix it. To answer your question, that was not supposed to happen. I will try to get it fixed this weekend :)

**Comment by dantecarlo** [comment]
I found the same bug,
this is my code!
`
  const [plateInput, setPlateInput] = useState('')

<CharacterInput
        placeHolder="-------"
        showCharBinary="1111111"
        handleChange={setPlateInput}
        inputType="contained"
  />
`

 thx

**Comment by dantecarlo** [comment]
A pseudo fix that I found is adding a 0 at the end of the showCharBinary.
If you need 6 characters your showCharBinary and placeHolder size should be 7.

```
const [plateInput, setPlateInput] = useState('')
<CharacterInput
        placeHolder="-------"
        showCharBinary="1111110"
        handleChange={setPlateInput}
        inputType="contained"
      />
```

**Comment by dantecarlo** [comment]
> If you fill the example inputs with random characters and then try to rewrite it with something else, the last character input is treated differently and you are unable to change the previously given value. Something might be wrong in the traverseInputs function.

A pseudo fix that I found is adding a 0 at the end of the showCharBinary.
If you need 6 characters your showCharBinary and placeHolder size should be 7.

```
const [plateInput, setPlateInput] = useState('')
<CharacterInput
        placeHolder="-------"
        showCharBinary="1111110"
        handleChange={setPlateInput}
        inputType="contained"
/>
```

**Comment by Kolos65** [comment]
Sorry if I wasn't clear enough, but the fix is to change this:
`const clearInputOnFocus = (inputPos: number) => {
    if (inputPos !== inputLength - 1) {
      singleInputRef[inputPos].clear();
      updateChangedChar(inputPos, '');
    }
  }`
to this:
`const clearInputOnFocus = (inputPos: number) => {
      singleInputRef[inputPos].clear();
      updateChangedChar(inputPos, '');
  }`

**Comment by hrutvikpatel** [comment]
Oh, yea that makes sense! Thank you lol for fixing my bugs :) I will make the change soon. Just got bare assignments :(

**Comment by hrutvikpatel** [comment]
@Kolos65 and @dantecarlo I published 1.0.4. Let me know if that works for you. If it does I will close this issue :)
